app/services/weights_and_measures.rb

```diff
@@ -26,9 +26,15 @@ def system
   def self.variant_unit_options
     available_units_sorted.flat_map do |measurement, measurement_info|
       measurement_info.filter_map do |scale, unit_info|
+        # Our code is based upon English based number formatting
+        # Some language locales like +hu+ uses a comma(,) for decimal separator
+        # While in English, decimal separator is represented by a period.
+        # e.g. en: 0.001, hu: 0,001
+        # Hence the results become "weight_0,001" for hu while or code recognizes "weight_0.001"
         scale_clean =
           ActiveSupport::NumberHelper.number_to_rounded(scale, precision: nil, significant: false,
```

Comment:
I believe this helper should only be used to round numbers when we want to display the output to the end user as it is highly influenced by the locale being used.

Upon further investigation (_from old UI_), I think it's being an alternative to Javascript's `parseFloat` method which behaves just like this helper. And here Rails being Rails, it just does so much under the hood that it feels like magic and these odd behaviors can be missed easily  😓